apps/supernova/src/App.jsx

```diff
@@ -62,13 +59,31 @@ function App(props = {}) {
   )
 }
 
+const AppWithOldUrlStructure = (props) => {
+  // syncs navigation relevant states with the url for deep links
+  // gets the state from the URL in the beginning
+  // sets the URL from state information
+  useUrlState()
+  return <App {...props} />
+}
+
+const AppWithNewUrlStructure = (props) => {
+  /**
+   * [TODO]
+   * move the URL state changes closer to the origins of the change
+   * so the whole app does not unnecessarily re-render.
+   */
+  useUrlQueryState()
+  return <App {...props} />
+}
+
 const StyledApp = (props) => {
   return (
     <AppShellProvider theme={`${props.theme ? props.theme : "theme-dark"}`}>
       {/* load appstyles inside the shadow dom */}
       <style>{styles.toString()}</style>
       <StoreProvider options={props}>
-        <App {...props} />
+        {props?.enableNewUrlStructure ? <AppWithNewUrlStructure {...props} /> : <AppWithOldUrlStructure {...props} />}
```

Comment:
@taymoor89 this can be enabled by changing the configuration we have in the preview for supernova and also activated per query param.

Reply:
I really like this feature switch ❤️